### PR TITLE
feat(server): handle authorization for updates

### PR DESCRIPTION
**Pull request by pedroscosta**
## Summary by CodeRabbit

- New Features
  - Adds granular authorization for updates with optional pre- and post-mutation checks, preventing unauthorized changes and ensuring only approved updates are applied. Successful updates now return the updated record.

- Tests
  - Introduces comprehensive test coverage for update authorization flows, including pre-/post-check success and failure scenarios, composite conditions, and edge cases.

### CODE REVIEW

File: packages/live-state/src/server/router.ts [hunk 1]
```diff
@@ -265,12 +269,40 @@ export class Route<
...
+        );
+        const authorized = applyWhere(
+          inferValue(result) as Record<string, any>,
+          authorizationWhereClause
+        );
+        if (!authorized) {
+          throw new Error("Not authorized");
+        }
+      }
+
```
[the remark this answers is not in the document]
Comment (pedroscosta) [reply packages/live-state/src/server/router.ts:303]: Fixed in the next PR
